src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/AbstractSearchAction.java

```diff
@@ -126,6 +135,26 @@ public RestResponse buildResponse(SearchResponse response) throws Exception {
 
     @Override
     public List<Route> routes() {
-        return ImmutableList.of(new Route(RestRequest.Method.POST, urlPath), new Route(RestRequest.Method.GET, urlPath));
+        List<Route> routes = new ArrayList<>();
+        for (String path : urlPaths) {
+            routes.add(new Route(RestRequest.Method.POST, path));
+            routes.add(new Route(RestRequest.Method.GET, path));
+        }
+        return routes;
+    }
+
+    @Override
+    public List<ReplacedRoute> replacedRoutes() {
```

Comment:
I think we will deprecate the old `opendistro` path in future. Should we put the old paths in `deprecatedRoutes` ? Haven't used the `replacedRoutes` and `deprecatedRoutes`. Just asking to understand the differences.

Reply:
Yeah the difference between them is that `deprecatedRoute` is only for routes which are deprecated but have no replacement. `replacedRoute` Is used for routes which are replaced with a new path while deprecating the old one.

Reply:
Got it. Will user see some warning if they use the old path?

Reply:
https://github.com/opensearch-project/opensearch-docs/blob/f1f6e7109f2b5312eb5dd47c4025b7373683db7a/server/src/main/java/org/elasticsearch/rest/RestHandler.java#L89

Reply:
Yes, you'll see a deprecation warning from [here](https://github.com/opensearch-project/opensearch-docs/blob/f1f6e7109f2b5312eb5dd47c4025b7373683db7a/server/src/main/java/org/elasticsearch/rest/RestController.java#L150).

Reply:
Nice. Thanks. Have no permission to check these two links shared by @dblock. But it's ok if we have such warning.